src/potions.ts

```diff
@@ -79,6 +81,26 @@ for (const effectGroup of mutuallyExclusiveList) {
   }
 }
 
+const allNoHookahEffects = Effect.all().filter((effect) => !effect.attributes.includes("nohookah"));
+const validWishString: Map<Effect, string> = new Map();
+function getValidWishString(effect: Effect): string | undefined {
```

Comment:
extremely funny thought: do we need to check these substrings against item names as well?

Reply:
update:
we only care about if a substring is an exact match for an available item name. This will probably? never happen.
But!
```
const availableItems = [...new Set(Location.all().filter((l) => canAdventure(l)).map((l) => getMonsters(l)).filter((m) => m.copyable).map((m) => itemDropsArray(m)).flat())];
```
or something like that

Reply:
I added this in, but we should also keep in mind that this is created at the instance of invoking garbo. We may gain (one-day tickets, transponder etc) and/or lose (content unlocker effects running out) access to locations as garbo runs, so this will then be a slightly incorrect list. 

(But I don't think we should be updating this every turn, so this is the best we can do)